src/main/java/com/direwolf20/buildinggadgets/common/schema/BGTemplate.java

```diff
@@ -0,0 +1,68 @@
+package com.direwolf20.buildinggadgets.common.schema;
+
+import com.google.common.collect.AbstractIterator;
+import it.unimi.dsi.fastutil.longs.Long2ObjectMap;
+import it.unimi.dsi.fastutil.longs.Long2ObjectMap.Entry;
+import it.unimi.dsi.fastutil.objects.ObjectIterator;
+import net.minecraft.block.BlockState;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.util.math.BlockPos.Mutable;
+import net.minecraft.util.math.Vec3i;
+
+import java.util.Iterator;
+
+public final class BGTemplate implements Iterable<MutableBlockData> {
+    private static final int BYTE_MASK_12BIT = 0xF_FF;
+    private static final int BYTE_MASK_14BIT = 0x3F_FF;
+    private final Long2ObjectMap<BlockState> states;
+    private final Long2ObjectMap<CompoundNBT> nbtData;
+    private final BoundingBox bounds;
+    private final String author;
+    private final String name;
+
+    private BGTemplate(Long2ObjectMap<BlockState> states, Long2ObjectMap<CompoundNBT> nbtData, BoundingBox bounds,
+                      String author, String name) {
+        this.states = states;
+        this.nbtData = nbtData;
+        this.bounds = bounds;
+        this.author = author;
+        this.name = name;
```

Comment:
Templates don't need names, this is something I disliked from the original implementation, this can just come out.

Reply:
Ok. Should they keep their author?

Reply:
Yeah, that bits fine :D